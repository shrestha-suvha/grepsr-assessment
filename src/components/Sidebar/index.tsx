import React, { FC, useState } from 'react';
import { AiFillHome, AiOutlinePlus } from 'react-icons/ai';
import { HiDatabase } from 'react-icons/hi';
import { BsThreeDotsVertical } from 'react-icons/bs';
import { BiChevronRight, BiChevronDown } from 'react-icons/bi';
import Collapse from 'react-bootstrap/Collapse';
import { Row, Col } from 'react-bootstrap';
import { FiSearch } from 'react-icons/fi';

const Sidebar: FC = () => {
  const [open, setOpen] = useState(false);
  const [addData, setAddData] = useState(false);
  return (
    <>
      <div className="side_nav">
        <div className="d-flex justify-content-between p-1 side_nav__item--active">
          <div className="ms-2">
            <AiFillHome size="1.1em" color="#1276ff" className="d-inline" />
            <span className="fw-bold text-info fs-esm ps-2">Sample data 1</span>
            <div className="fs-esm ps-4 mt-1 text-muted">105 K records</div>
          </div>
          <div className="pt-2 me-2">
            <BsThreeDotsVertical color="gray" className="side_nav__icon--active" />
          </div>
        </div>
        <div className="d-flex justify-content-between align-items-center border-bottom pb-2 mt-4">
          <span>
            <BiChevronRight size={'1em'} className="side_nav__icon--expand" />
            <span className="side_nav__item-title ps-1 ">Sample data</span>
          </span>
          <AiOutlinePlus className="side_nav__icon--add" id="dropdown-basic" onClick={() => setAddData(!addData)} />
        </div>

        <div className="side_nav__item--empty d-flex flex-column justify-content-center align-items-center">
          <div>
            <HiDatabase className="side_nav__icon--empty" />
          </div>
          <div className="pt-2">No sample data</div>
        </div>
        <div className="d-flex justify-content-between align-items-center border-bottom pb-2 mt-4">
          <span>
            <BiChevronRight size={'1em'} className="side_nav__icon--expand" />
            <span className="side_nav__item-title ps-1 ">Dataset 2 </span>
          </span>
          <AiOutlinePlus className="side_nav__icon--add" />
        </div>
        <div className="d-flex justify-content-between align-items-center border-bottom pb-2 mt-4">
          <span>
            {!open ? (
              <BiChevronRight size={'1em'} className="side_nav__icon--expand" onClick={() => setOpen(!open)} />
            ) : (
              <BiChevronDown size={'1em'} className="side_nav__icon--expand" onClick={() => setOpen(!open)} />
            )}
            <span className="side_nav__item-title ps-1 ">Dataset 3</span>
          </span>
        </div>
        <Collapse in={open}>
          <div className="example-collapse-text">
            <Row className="mb-2">
              <Col className="input-group input-group-sm ms-2 mt-3">
                <span className="input-group-text" style={{ backgroundColor: '#F5F6F9' }}>
                  <FiSearch size="1.5em" color="gray" />
                </span>
                <input
                  type="text"
                  className="form-control border-start-0 top-nav__search me-2"
                  placeholder="Search"
                  aria-label="Search"
                  style={{ backgroundColor: '#F5F6F9' }}
                />
              </Col>
            </Row>
            <Row className="side_nav__item--inactive mt-5">
              <Col>
                <span className="fw-bold fs-esm ">Dataset 1</span>
                <div className="fs-esm mt-1 text-muted">105 K records</div>
              </Col>
              <Col className="d-flex justify-content-end mt-3 me-2">
                <BsThreeDotsVertical color="gray" />
              </Col>
            </Row>
            <Row className="side_nav__item--inactive">
              <Col>
                <span className="fw-bold fs-esm">Dataset 2</span>
                <div className="fs-esm mt-1 text-muted ">235 K records</div>
              </Col>
              <Col className="d-flex justify-content-end mt-3 me-2">
                <BsThreeDotsVertical color="gray" />
              </Col>
            </Row>
            <Row className="side_nav__item--inactive">
              <Col>
                <span className="fw-bold fs-esm">Dataset 3</span>
                <div className="fs-esm mt-1 text-muted">235 K records</div>
              </Col>
              <Col className="d-flex justify-content-end mt-3 me-2">
                <BsThreeDotsVertical color="gray" />
              </Col>
            </Row>
          </div>
        </Collapse>
      </div>
    </>
  );
};
export default Sidebar;
