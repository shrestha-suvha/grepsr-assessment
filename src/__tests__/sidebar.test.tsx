import { render, screen, cleanup } from '@testing-library/react';
import { create } from 'react-test-renderer';

import Sidebar from 'src/components/Sidebar';

afterEach(() => {
  cleanup();
});

test('should render Sidebar component', () => {
  render(<Sidebar />);
  const SidebarElement = screen.getByTestId('Sidebar-1');
  const noSampleData = screen.getByText('No sample data');
  expect(SidebarElement).toBeInTheDocument();
  expect(noSampleData).toBeInTheDocument();
});

test('matches snapshot', () => {
  const tree = create(<Sidebar />).toJSON();
  expect(tree).toMatchSnapshot();
});
