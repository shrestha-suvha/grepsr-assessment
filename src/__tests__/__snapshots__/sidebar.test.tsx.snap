// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="side_nav "
  data-testid="Sidebar-1"
>
  <div
    className="d-flex justify-content-between p-1 side_nav__item--active"
  >
    <div
      className="ms-2"
    >
      <svg
        className="d-inline"
        color="#1276ff"
        fill="currentColor"
        height="1.1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": "#1276ff",
          }
        }
        viewBox="0 0 1024 1024"
        width="1.1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M946.5 505L534.6 93.4a31.93 31.93 0 0 0-45.2 0L77.5 505c-12 12-18.8 28.3-18.8 45.3 0 35.3 28.7 64 64 64h43.4V908c0 17.7 14.3 32 32 32H448V716h112v224h265.9c17.7 0 32-14.3 32-32V614.3h43.4c17 0 33.3-6.7 45.3-18.8 24.9-25 24.9-65.5-.1-90.5z"
        />
      </svg>
      <span
        className="fw-bold text-info fs-esm ps-2"
      >
        Sample data 1
      </span>
      <div
        className="fs-esm ps-4 mt-1 text-muted"
      >
        105 K records
      </div>
    </div>
    <div
      className="pt-2 me-2"
    >
      <svg
        className="side_nav__icon--active"
        color="gray"
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": "gray",
          }
        }
        viewBox="0 0 16 16"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M9.5 13a1.5 1.5 0 1 1-3 0 1.5 1.5 0 0 1 3 0zm0-5a1.5 1.5 0 1 1-3 0 1.5 1.5 0 0 1 3 0zm0-5a1.5 1.5 0 1 1-3 0 1.5 1.5 0 0 1 3 0z"
        />
      </svg>
    </div>
  </div>
  <div
    className="d-flex justify-content-between align-items-center border-bottom pb-2 mt-4"
  >
    <span>
      <svg
        className="side_nav__icon--expand"
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M10.707 17.707 16.414 12l-5.707-5.707-1.414 1.414L13.586 12l-4.293 4.293z"
        />
      </svg>
      <span
        className="side_nav__item-title ps-1 "
      >
        Sample data
      </span>
    </span>
    <svg
      className="side_nav__icon--add"
      fill="currentColor"
      height="1em"
      id="dropdown-basic"
      onClick={[Function]}
      stroke="currentColor"
      strokeWidth="0"
      style={
        Object {
          "color": undefined,
        }
      }
      t="1551322312294"
      version="1.1"
      viewBox="0 0 1024 1024"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <defs />
      <path
        d="M474 152m8 0l60 0q8 0 8 8l0 704q0 8-8 8l-60 0q-8 0-8-8l0-704q0-8 8-8Z"
      />
      <path
        d="M168 474m8 0l672 0q8 0 8 8l0 60q0 8-8 8l-672 0q-8 0-8-8l0-60q0-8 8-8Z"
      />
    </svg>
  </div>
  <div
    className="side_nav__item--empty d-flex flex-column justify-content-center align-items-center"
  >
    <div>
      <svg
        aria-hidden="true"
        className="side_nav__icon--empty"
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 20 20"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M3 12v3c0 1.657 3.134 3 7 3s7-1.343 7-3v-3c0 1.657-3.134 3-7 3s-7-1.343-7-3z"
        />
        <path
          d="M3 7v3c0 1.657 3.134 3 7 3s7-1.343 7-3V7c0 1.657-3.134 3-7 3S3 8.657 3 7z"
        />
        <path
          d="M17 5c0 1.657-3.134 3-7 3S3 6.657 3 5s3.134-3 7-3 7 1.343 7 3z"
        />
      </svg>
    </div>
    <div
      className="pt-2"
    >
      No sample data
    </div>
  </div>
  <div
    className="d-flex justify-content-between align-items-center border-bottom pb-2 mt-4"
  >
    <span>
      <svg
        className="side_nav__icon--expand"
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M10.707 17.707 16.414 12l-5.707-5.707-1.414 1.414L13.586 12l-4.293 4.293z"
        />
      </svg>
      <span
        className="side_nav__item-title ps-1 "
      >
        Dataset 2 
      </span>
    </span>
    <svg
      className="side_nav__icon--add"
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      strokeWidth="0"
      style={
        Object {
          "color": undefined,
        }
      }
      t="1551322312294"
      version="1.1"
      viewBox="0 0 1024 1024"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <defs />
      <path
        d="M474 152m8 0l60 0q8 0 8 8l0 704q0 8-8 8l-60 0q-8 0-8-8l0-704q0-8 8-8Z"
      />
      <path
        d="M168 474m8 0l672 0q8 0 8 8l0 60q0 8-8 8l-672 0q-8 0-8-8l0-60q0-8 8-8Z"
      />
    </svg>
  </div>
  <div
    className="d-flex justify-content-between align-items-center border-bottom pb-2 mt-4"
  >
    <span>
      <svg
        className="side_nav__icon--expand"
        fill="currentColor"
        height="1em"
        onClick={[Function]}
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M10.707 17.707 16.414 12l-5.707-5.707-1.414 1.414L13.586 12l-4.293 4.293z"
        />
      </svg>
      <span
        className="side_nav__item-title ps-1 "
      >
        Dataset 3
      </span>
    </span>
  </div>
  <div
    aria-expanded={null}
    className="example-collapse-text collapse"
  >
    <div
      className="mb-2 row"
    >
      <div
        className="input-group input-group-sm ms-2 mt-3 col"
      >
        <span
          className="input-group-text"
          style={
            Object {
              "backgroundColor": "#F5F6F9",
            }
          }
        >
          <svg
            color="gray"
            fill="none"
            height="1.5em"
            stroke="currentColor"
            strokeLinecap="round"
            strokeLinejoin="round"
            strokeWidth="2"
            style={
              Object {
                "color": "gray",
              }
            }
            viewBox="0 0 24 24"
            width="1.5em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <circle
              cx="11"
              cy="11"
              r="8"
            />
            <line
              x1="21"
              x2="16.65"
              y1="21"
              y2="16.65"
            />
          </svg>
        </span>
        <input
          aria-label="Search"
          className="form-control border-start-0 top-nav__search me-2"
          placeholder="Search"
          style={
            Object {
              "backgroundColor": "#F5F6F9",
            }
          }
          type="text"
        />
      </div>
    </div>
    <div
      className="side_nav__item--inactive mt-5 row"
    >
      <div
        className="col"
      >
        <span
          className="fw-bold fs-esm "
        >
          Dataset 1
        </span>
        <div
          className="fs-esm mt-1 text-muted"
        >
          105 K records
        </div>
      </div>
      <div
        className="d-flex justify-content-end mt-3 me-2 col"
      >
        <svg
          color="gray"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": "gray",
            }
          }
          viewBox="0 0 16 16"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M9.5 13a1.5 1.5 0 1 1-3 0 1.5 1.5 0 0 1 3 0zm0-5a1.5 1.5 0 1 1-3 0 1.5 1.5 0 0 1 3 0zm0-5a1.5 1.5 0 1 1-3 0 1.5 1.5 0 0 1 3 0z"
          />
        </svg>
      </div>
    </div>
    <div
      className="side_nav__item--inactive row"
    >
      <div
        className="col"
      >
        <span
          className="fw-bold fs-esm"
        >
          Dataset 2
        </span>
        <div
          className="fs-esm mt-1 text-muted "
        >
          235 K records
        </div>
      </div>
      <div
        className="d-flex justify-content-end mt-3 me-2 col"
      >
        <svg
          color="gray"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": "gray",
            }
          }
          viewBox="0 0 16 16"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M9.5 13a1.5 1.5 0 1 1-3 0 1.5 1.5 0 0 1 3 0zm0-5a1.5 1.5 0 1 1-3 0 1.5 1.5 0 0 1 3 0zm0-5a1.5 1.5 0 1 1-3 0 1.5 1.5 0 0 1 3 0z"
          />
        </svg>
      </div>
    </div>
    <div
      className="side_nav__item--inactive row"
    >
      <div
        className="col"
      >
        <span
          className="fw-bold fs-esm"
        >
          Dataset 3
        </span>
        <div
          className="fs-esm mt-1 text-muted"
        >
          235 K records
        </div>
      </div>
      <div
        className="d-flex justify-content-end mt-3 me-2 col"
      >
        <svg
          color="gray"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": "gray",
            }
          }
          viewBox="0 0 16 16"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M9.5 13a1.5 1.5 0 1 1-3 0 1.5 1.5 0 0 1 3 0zm0-5a1.5 1.5 0 1 1-3 0 1.5 1.5 0 0 1 3 0zm0-5a1.5 1.5 0 1 1-3 0 1.5 1.5 0 0 1 3 0z"
          />
        </svg>
      </div>
    </div>
  </div>
</div>
`;
